pkg/network/ebpf/c/protocols/http2/decoding.h

```diff
@@ -117,29 +129,35 @@ static __always_inline void parse_field_indexed(dynamic_table_index_t *dynamic_i
 
 READ_INTO_BUFFER(path, HTTP2_MAX_PATH_LEN, BLK_SIZE)
 
-// parse_field_literal handling the case when the key is part of the static table and the value is a dynamic string
-// which will be stored in the dynamic table.
+// parse_field_literal parses headers with the header name being in the static
+// table and the value is a literal string to be stored in the dynamic table.
+//
+// We are only interested in path headers, and will skip those that are not.
 static __always_inline bool parse_field_literal(struct __sk_buff *skb, skb_info_t *skb_info, http2_header_t *headers_to_process, __u64 index, __u64 global_dynamic_counter, __u8 *interesting_headers_counter) {
     __u64 str_len = 0;
-    if (!read_var_int(skb, skb_info, MAX_6_BITS, &str_len)) {
+    if (!read_hpack_int(skb, skb_info, MAX_6_BITS, &str_len)) {
         return false;
     }
-    // The key is new and inserted into the dynamic table. So we are skipping the new value.
 
+    // The header name is new and inserted in the dynamic table - we skip the new value.
     if (index == 0) {
         skb_info->data_off += str_len;
         str_len = 0;
-        if (!read_var_int(skb, skb_info, MAX_6_BITS, &str_len)) {
+        if (!read_hpack_int(skb, skb_info, MAX_6_BITS, &str_len)) {
             return false;
         }
         goto end;
     }
+
+    // We skip if:
+    // - The string is too big
+    // - This is not a path
+    // - We won't be able to store the header info
     if (str_len > HTTP2_MAX_PATH_LEN || index != kIndexPath || headers_to_process == NULL) {
         goto end;
     }
 
-    __u32 final_size = str_len < HTTP2_MAX_PATH_LEN ? str_len : HTTP2_MAX_PATH_LEN;
-    if (skb_info->data_off + final_size > skb_info->data_end) {
```

Comment:
IIRC, `final_size` was meant to pass the verifier

Reply:
Hmm 🤔 I did of course build and run `system-probe` before committing, and the CI seems to be passing without issues. Maybe the verifier was somehow unhappy about something else that has since changed?

Reply:
As long as the CI is happy, I'm happy